### ISSUE TITLE
Does DDPG work on a cuda device with torch?[rllib]

### ISSUE THREAD

**Issue by FabianSchuetze** [ray-project/ray#10889]
I receive several placement device errors when I try to train a model `DDPG` with the `torch` framework.

I installed the latest nighty build (ray-0.9.0.dev0-cp38-cp38-manylinux1_x86_64) and also merged the changes described in  #9683 . The config I used to instantiate the model is:

```Python
    config['num_gpus'] = 1.0
    config['framework'] = 'torch'
    trainer = ddpg.DDPGTrainer(env='LunarLanderContinuous-v2', config=config)
    trainer.train()

```
However, already model in the first lines in the `ddpg_actor_critic_loss`function 
https://github.com/ray-project/ray/blob/41d7221dcd45016b10140cf7f120fe5f614548c8/rllib/agents/ddpg/ddpg_torch_policy.py#L48
is already not placed on the GPU. The model trains well when using only CPUs but fails when I specify the 'num_gpu' option.

Does anybody else have experience with placing the driver on the GPU successfully? I wonder if the question is in fact a duplicate of #10271, as I do not call the train functionality via tue. Does somebody have an idea what I need to do to place the driver model on the GPU?

**Comment by ericl** [comment]
I think https://github.com/ray-project/ray/pull/10834/files fixes this. Please re-open this issue if there is still a problem after that commit.

**Comment by FabianSchuetze** [comment]
Thanks for the reply, @ericl !  

Besides #10834 , I also needed to merge (the `torch_model.py` file from) #9683 . #10834 works for other agents I have tried (SAC).

**Comment by sven1977** [comment]
Yes, for some reason, we never merged the GPU DDPG torch fixes. This will be done now in this PR, which fixes a couple of other issues as well for DDPG and SAC (related to e.g. Conv2D state preprocessors).

https://github.com/ray-project/ray/issues/10765

Also has a small SAC discrete loss fix.

**Comment by FabianSchuetze** [comment]
Thanks for the info!

**Comment by sven1977** [comment]
Hey @FabianSchuetze , also this PR is merged right now, which fixes the old GPU problem with DDPG (eventually; sorry this took so long!).
https://github.com/ray-project/ray/pull/11033

**Comment by FabianSchuetze** [comment]
Thanks @sven1977 the the info! Its great to hear the RP is merged & congratulations to the 1.0 release!
